api/src/main/java/org/apache/cloudstack/api/command/admin/router/GetRouterHealthCheckResultsCmd.java

```diff
@@ -0,0 +1,114 @@
+import com.cloud.network.router.VirtualRouter;
+import com.cloud.user.Account;
+import com.cloud.vm.VirtualMachine;
+
+@APICommand(name = GetRouterHealthCheckResultsCmd.APINAME,
+        responseObject = DomainRouterHealthCheckResultsResponse.class,
+        description = "Starts a router.",
+        entityType = {VirtualMachine.class},
+        requestHasSensitiveInfo = false,
+        responseHasSensitiveInfo = false,
+        since = "4.13.1")
+public class GetRouterHealthCheckResultsCmd extends BaseCmd {
+    public static final Logger s_logger = Logger.getLogger(GetRouterHealthCheckResultsCmd.class.getName());
+    public static final String APINAME = "getRouterHealthCheckResults";
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+
+    @Parameter(name = ApiConstants.ID, type = CommandType.UUID, entityType = DomainRouterResponse.class,
+            required = true, description = "the ID of the router")
+    private Long id;
+
+    @Parameter(name = "performfreshchecks", type = CommandType.BOOLEAN, description = "if true is passed for this parameter, " +
+            "health checks are performed on the fly. Else last performed checks data is fetched")
+    private Boolean performFreshChecks;
+
+    /////////////////////////////////////////////////////
+    /////////////////// Accessors ///////////////////////
+    /////////////////////////////////////////////////////
+
+    public Long getId() {
+        return id;
+    }
+
+    public boolean shouldPerformFreshChecks() {
+        return performFreshChecks == null ? false : performFreshChecks.booleanValue();
+    }
+
+    /////////////////////////////////////////////////////
+    /////////////// API Implementation///////////////////
+    /////////////////////////////////////////////////////
+
+    @Override
+    public String getCommandName() {
+        return APINAME.toLowerCase() + BaseCmd.RESPONSE_SUFFIX;
+    }
+
+    @Override
+    public long getEntityOwnerId() {
+        VirtualRouter router = _entityMgr.findById(VirtualRouter.class, getId());
+        if (router != null) {
+            return router.getAccountId();
+        }
+
+        return Account.ACCOUNT_ID_SYSTEM;
+    }
+
+    @Override
+    public void execute() throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
+        CallContext.current().setEventDetails("Router Id: " + this._uuidMgr.getUuid(VirtualMachine.class, getId()));
+        VirtualRouter router = _routerService.findRouter(getId());
+        Map<String, String> result = null;
+        if (router == null || router.getRole() != VirtualRouter.Role.VIRTUAL_ROUTER) {
+            throw new InvalidParameterValueException("Can't find router by id");
+        } else {
+            result = _routerService.getRouterHealthCheckResults(getId(), shouldPerformFreshChecks());
```

Comment:
This can be executed without an else block and won't need to split the result definition into multiple lines

Reply:
Done